tests-legacy/check_file_syntax.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-! (find . -name "*.php" ! -path "./vendor/*" ! -path "./tools/*" -print0 | xargs -0 -n1 -P4 php -l | grep -q "Parse error")
+! (find . -name "*.php" ! -path "./vendor/*" ! -path "./tools/*" ! -path "./modules/*" -print0 | xargs -0 -n1 -P4 php -l | grep -q "Parse error")
```

Comment:
Why avoiding this folder? As we do not check the syntax of our modules in each repository, that was a nice backup.

Reply:
because we didn't do it before, so I don't want to add a "new" check on this pull request.

I agree to do it, but in another pull request (what will happens if a vendor file from a module break the core build?).

Reply:
Can't find a compromise by removing `vendor` folders recursively?

Reply:
It is maybe the time to add a syntax checker in all our modules?

Reply:
I guess we have no choice!